cpp/include/cudf/io/parquet.hpp

```diff
@@ -859,82 +860,84 @@ class chunked_parquet_writer_options_builder {
 };
 
 /**
- * @brief Forward declaration of anonymous chunked-writer state struct.
+ * @brief Merges multiple raw metadata blobs that were previously created by write_parquet
+ * into a single metadata blob
+ *
+ * @ingroup io_writers
+ *
+ * @param[in] metadata_list List of input file metadata
+ * @return A parquet-compatible blob that contains the data for all rowgroups in the list
  */
-struct pq_chunked_state;
+std::unique_ptr<std::vector<uint8_t>> merge_rowgroup_metadata(
+  const std::vector<std::unique_ptr<std::vector<uint8_t>>>& metadata_list);
 
 /**
- * @brief Begin the process of writing a parquet file in a chunked/stream form.
+ * @brief chunked parquet writer class to handle options and write tables in chunks.
  *
- * The intent of the write_parquet_chunked_ path is to allow writing of an
+ * The intent of the parquet_chunked_writer is to allow writing of an
  * arbitrarily large / arbitrary number of rows to a parquet file in multiple passes.
  *
  * The following code snippet demonstrates how to write a single parquet file containing
  * one logical table by writing a series of individual cudf::tables.
+ *
  * @code
  *  ...
  *  std::string filepath = "dataset.parquet";
  *  cudf::io::chunked_parquet_writer_options options =
  *  cudf::io::chunked_parquet_writer_options::builder(cudf::sink_info(filepath), table->view());
  *  ...
- *  auto state = cudf::write_parquet_chunked_begin(options);
- *    cudf::write_parquet_chunked(table0, state);
- *    cudf::write_parquet_chunked(table1, state);
- *    ...
- *  cudf_write_parquet_chunked_end(state);
- * @endcode
- *
- * @param[in] options Settings for controlling writing behavior.
- * @param[in] mr Device memory resource to use for device memory allocation.
- *
- * @return pointer to an anonymous state structure storing information about the chunked write.
- * this pointer must be passed to all subsequent write_parquet_chunked() and
- * write_parquet_chunked_end() calls.
+ *  cudf::io::parquet_chunked_writer writer(options)
+ *  writer.write(table0)
+ *  writer.write(table1)
+ *  ...
+ *  writer.write_end()
+ *  @endcode
  */
-std::shared_ptr<pq_chunked_state> write_parquet_chunked_begin(
-  chunked_parquet_writer_options const& options,
-  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+class parquet_chunked_writer {
```

Comment:
Where's the destructor?

Reply:
Don't need a destructor.

Reply:
Don't we need one to call write_end?

Reply:
I have added it in impl so even detail can use it https://github.com/rapidsai/cudf/blob/6fd27e5bfd6427a7a3133df42dea56f473716938/cpp/src/io/parquet/writer_impl.cu#L670